File: AspNetMvcUnleashed/0672329980_source/Chapter12/VB/MvcApplication1/MvcApplication1/Controllers/WindowsController.vb
Public Class WindowsController
    Inherits Controller

    <Authorize(Users:="redmond\swalther")> _
    Public Function Index() As ActionResult
        ViewData("userName") = User.Identity.Name
        Return View()
    End Function

    <Authorize(Roles:="Managers")> _
    Public Function SalesFigures() As ActionResult
        ViewData("userName") = User.Identity.Name
        Return View()
    End Function

    Public Function SecretStuff() As ActionResult
        If User.IsInRole("Managers") Then
            Return View()
        End If

        Return New HttpUnauthorizedResult()
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter12/CS/MvcApplication1/MvcApplication1/Controllers/WindowsController.cs
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class WindowsController : Controller
    {
        [Authorize(Users="redmond\\swalther")]
        public ActionResult Index()
        {
            ViewData["userName"] = User.Identity.Name;
            return View();
        }

        [Authorize(Roles = "Managers")]
        public ActionResult SalesFigures()
        {
            ViewData["userName"] = User.Identity.Name;
            return View();
        }

        public ActionResult SecretStuff()
        {
            if (User.IsInRole("Managers"))
                return View();

            return new HttpUnauthorizedResult();
        }



    }
}
